### DISCUSSION TITLE
Design compare feature

### DISCUSSION THREAD

**Discussion by johannadm** [netwerk-digitaal-erfgoed/solid-crs#571] [Project Erfgoedpod]
Hi all, 

For the compare feature, as discussed in today's meeting, we suggest the following design as a special page in the presentation module:
![image](https://user-images.githubusercontent.com/80322390/153198138-6c3ad141-1fc8-40db-9412-b0982031d96f.png)

You would reach this page by clicking on 'aangepast object bekijken' in this view of the object detail page:
![image](https://user-images.githubusercontent.com/80322390/153198448-eadbf5a9-a74c-440e-aa2a-c7fff2986f3b.png)

Let us know what you think of this suggestion!

**Comment by EnnoMeijers** [comment]
Thank you for the proposal, nice that the comparison can between the original and the annotated object can be supported. Would it be possible to make it more clear visually which metadata belongs to the original object and which to the annotated object? And I suppose the left bar navigation will give the opportunity to go back the 'Bruikleen' page?

**Reply by lem-onade** [reply]
> Would it be possible to make it more clear visually which metadata belongs to the original object and which to the annotated object? 

I like this, we weren't sure how to distinguish between old/new data

> And I suppose the left bar navigation will give the opportunity to go back the 'Bruikleen' page?

The first screenshot is of the presentation app, would be a new feature where two objects can be compared, which would only be accessible from the link at the bottom of the second screenshot. Clicking the link could open a new tab or navigate to the comparison page in the presentation app. Going back could then be done by closing the tab or going back.
Perhaps, in the future, this comparison feature can be useful for other features as well.

**Comment by mielvds** [comment]
I'm definitely ok with this design and I also agree with Enno. A highlighted background and a "From <WebID/Label>" would suffice. Is the cell blank when a value is missing (eg. Institution 2 hasn't added a description of its own)?

**Reply by lem-onade** [reply]
>  Is the cell blank when a value is missing (eg. Institution 2 hasn't added a description of its own)?

I had in mind that the old value would be shown in this case (see objectnummer, type), but agree it could be better to just show the new values.

**Reply by mielvds** [reply]
If you do decide to show both, you should highlight those that deviate (like in a diff)
